server/plugin.go

```diff
@@ -318,18 +322,19 @@ func (p *Plugin) startNewCallThread(userID, channelID string, startAt int64, tit
 			return nil, fmt.Errorf("call is missing from channel state")
 		}
 
-		state.Call.ThreadID = createdPost.Id
+		state.Call.PostID = createdPost.Id
+		state.Call.ThreadID = threadID
```

Comment:
Technically this could be optimized since a non-root post would link to its root so we could likely store one field only but I feel it may become more confusing. Happy to hear thoughts though.

Reply:
The only thing that comes to mind is if wrangler or something moves the post?

Reply:
Good thought. I guess the edge case would be for that to happen with an on-going call which would likely mess some things. Let me have another pass and see what can be done.

Reply:
After thinking a bit about this I am not sure we can do much about this king of edge cases, not easily anyway. Moving a thread means deleting the original so if that happens during a call it would still mess things up since we don't really get notified that the post/thread has been deleted. 

To deal with this we'd probably have to implement a new hook on the server that triggers when a post gets deleted and react accordingly, likely preventing the deletion if a call is on-going since anything else would be even more complex/unexpected (e.g. dropping people from the call).

Reply:
> To deal with this we'd probably have to implement a new hook on the server that triggers when a post gets deleted and react accordingly

Who's working on internalizing wrangler? Cause this seems like an edge-case that we should probably handle, and a "PostWillBeMoved" hook would be really great to add. We could even do that. I'll add a ticket to track it, just in case: https://mattermost.atlassian.net/browse/MM-50408

Reply:
Yes, we need two hooks most likely, a deleted one and a moved one. But it could get tricky with recordings as they could finish after a call has ended so the mapping between previous post id and new post id requires some care.

Also please consider linking or closing https://mattermost.atlassian.net/browse/MM-50397 as I had a similar instinct :)

Reply:
Hah, that's funny. I closed mine and added a comment to yours. :)